Gems/Atom/RPI/Code/Source/RPI.Private/RPISystemComponent.cpp

```diff
@@ -100,24 +100,36 @@ namespace AZ
 
             auto settingsRegistry = AZ::SettingsRegistry::Get();
             const char* settingPath = "/O3DE/Atom/RPI/Initialization";
-
-            // if the command line contains -NullRenderer, merge it to setting registry
-            const char* nullRendererOption = "NullRenderer"; // command line option name
             const char* setregName = "NullRenderer"; // same as serialization context name for RPISystemDescriptor::m_isNullRenderer
-            const AzFramework::CommandLine* commandLine = nullptr;
-            AzFramework::ApplicationRequests::Bus::BroadcastResult(commandLine, &AzFramework::ApplicationRequests::GetApplicationCommandLine);
 
-            AZStd::string commandLineValue;
-            if (commandLine)
+            AZ::ApplicationTypeQuery appType;
+            ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationBus::Events::QueryApplicationType, appType);
+            
+            bool isNullRenderer = false;
+            if (appType.IsHeadless())
+            {
+                // if the application is `headless`, merge `NullRenderer` attribute to the setting registry
+                isNullRenderer = true;
+            }
```

Comment:
Shouldn't this also set -rhi=null somewhere?

Reply:
Not really needed, this supercedes it

Reply:
How so? From everything I've seen, they're independent switches. -NullRenderer controls the RPI layer only, and -rhi=null controls the RHI layer only. To turn off rendering completely, you need to pass both. Otherwise, I believe the RHI level will still try to initialize with DX12, Vulkan, etc even if no RPI features then use it.

Reply:
sorry, my mistake, you are right, this supercedes the block immediately below it
```
                // Otherwise if the command line contains -NullRenderer merge it to setting registry
                const char* nullRendererOption = "NullRenderer"; // command line option name
                const AzFramework::CommandLine* commandLine = nullptr;
                AzFramework::ApplicationRequests::Bus::BroadcastResult(commandLine, &AzFramework::ApplicationRequests::GetApplicationCommandLine);
                if (commandLine->GetNumSwitchValues(nullRendererOption) > 0)
                {
                    isNullRenderer = true;
                }
                // Otherwise if the command line contains -NullRenderer merge it to setting registry
                const char* nullRendererOption = "NullRenderer"; // command line option name
                const AzFramework::CommandLine* commandLine = nullptr;
                AzFramework::ApplicationRequests::Bus::BroadcastResult(commandLine, &AzFramework::ApplicationRequests::GetApplicationCommandLine);
                if (commandLine->GetNumSwitchValues(nullRendererOption) > 0)
                {
                    isNullRenderer = true;
                }
```
The the `-rhi=null` isn't really necessary because IIRC, making the RHI aliases for Vulkan, DX12, Metal for the new `HeadlessServer` cmake variants pointing to nothing, it will actually exclude the dependencies on those RHI modules and they won't launch (when building headless server launcher).